federation/src/main/scala/caliban/federation/FederationSupport.scala

```diff
@@ -81,9 +80,10 @@ abstract class FederationSupport(
        */
       override def resolve(value: _Entity): Step[R] =
         _entityMap
-          .get(value.__typename)
-          .fold[Step[R]](Step.NullStep)(resolver => QueryStep(resolver.resolve(value.value)))
-
+          .getOrElse(value.__typename, null) match {
+          case null     => Step.NullStep
+          case resolver => resolver.resolve(value.value)
```

Comment:
This preserves the behavior which should be fine for this PR but I just realized the null case here is more of a defect since it means either the gateway or the entity resolver map is wrong. I'll look into how Apollo does this in subgraphs. It's not part of the spec which makes me think it's undefined behavior that we should perhaps throw to avoid silent errors